checks.d/pgbouncer.py

```diff
@@ -0,0 +1,166 @@
+        'descriptors': [
+            ('database', 'db'),
+            ('user', 'user'),
+        ],
+        'metrics': {
+            'cl_active':            ('pgbouncer.pools.cl_active', GAUGE),
+            'cl_waiting':           ('pgbouncer.pools.cl_waiting', GAUGE),
+            'sv_active':            ('pgbouncer.pools.sv_active', GAUGE),
+            'sv_idle':              ('pgbouncer.pools.sv_idle', GAUGE),
+            'sv_used':              ('pgbouncer.pools.sv_used', GAUGE),
+            'sv_tested':            ('pgbouncer.pools.sv_tested', GAUGE),
+            'sv_login':             ('pgbouncer.pools.sv_login', GAUGE),
+            'maxwait':              ('pgbouncer.pools.maxwait', GAUGE),
+        },
+        'query': """SHOW POOLS""",
+    }
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+        self.dbs = {}
+
+    def _collect_stats(self, key, db, instance_tags):
+        """Query pgbouncer for various metrics
+        """
+
+        metric_scope = [self.STATS_METRICS, self.POOLS_METRICS]
+
+        try:
+            cursor = db.cursor()
+            for scope in metric_scope:
+
+                cols = scope['metrics'].keys()
+
+                try:
+                    query = scope['query']
+                    self.log.debug("Running query: %s" % query)
+                    cursor.execute(query)
+
+                    results = cursor.fetchall()
+                except pg.Error, e:
+                    self.log.warning("Not all metrics may be available: %s" % str(e))
+                    continue
+
+                for row in results:
+                    if row[0] == 'pgbouncer':
+                        continue
+
+                    desc = scope['descriptors']
+                    assert len(row) == len(cols) + len(desc)
+
+                    tags = [t for t in instance_tags]
+                    tags += ["%s:%s" % (d[0][1], d[1]) for d in zip(desc, row[:len(desc)])]
+
+                    values = zip([scope['metrics'][c] for c in cols], row[len(desc):])
+
+                    [v[0][1](self, v[0][0], v[1], tags=tags) for v in values]
+
+            if not results:
+                self.warning('No results were found for query: "%s"' % query)
+
+            cursor.close()
+        except pg.Error, e:
+            self.log.error("Connection error: %s" % str(e))
+            raise ShouldRestartException
+
+    def _get_connection(self, key, host, port, user, password, dbname, use_cached=True):
+        "Get and memoize connections to instances"
+        if key in self.dbs and use_cached:
+            return self.dbs[key]
+
+        elif host != "" and user != "":
+            try:
+                service_check_tags = [
+                    "host:%s" % host,
+                    "port:%s" % port
+                ]
+                if dbname:
+                    service_check_tags.append("db:%s" % dbname)
```

Comment:
Looks like it's redundant to have a `db:pgbouncer` tag when the service check is already about pgbouncer?